app/screens/home/channel_list/categories_list/categories/categories.test.tsx

```diff
@@ -26,3 +35,37 @@ describe('components/channel_list/categories', () => {
         expect(wrapper.toJSON()).toBeTruthy();
     });
 });
+
+describe('performance metrics', () => {
+    let database: Database;
+    const serverUrl = 'http://www.someserverurl.com';
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase(serverUrl);
+        database = server.database;
+    });
+
+    afterAll(async () => {
+        await TestHelper.tearDown();
+    });
+
+    it('properly send metric on load', () => {
+        renderWithEverything(<Categories/>, {database, serverUrl});
+        expect(PerformanceMetricsManager.endMetric).toHaveBeenCalledWith('mobile_team_switch', serverUrl);
+    });
+
+    it('properly call again after switching teams', async () => {
+        renderWithEverything(<Categories/>, {database, serverUrl});
+        expect(PerformanceMetricsManager.endMetric).toHaveBeenCalledTimes(1);
+        await act(async () => {
+            DeviceEventEmitter.emit(Events.TEAM_SWITCH, true);
+            await TestHelper.wait(100);
```

Comment:
Instead of waiting for 100 milliseconds, does mobile support Testing Library's `waitFor`? I've been really liking using that on the web for testing async code